src/packagedcode/rubygems.py

```diff
@@ -648,92 +628,172 @@ def normalize(gem_data, known_fields=known_fields):
 
 
 def parse_spec(location):
-    pass
+    """
+    Returns a dictionary containing .gemspec file data.
+    """
+    file_data = []
+    with io.open(location, "r", encoding="utf-8") as f:
+        file_data = f.read().splitlines()
+
+    dependencies = OrderedDict()
+
+    # defaults fields of .gemspec file
+    gemspec_data = {
+        "name": None,
+        "version": None,
+        "authors": None,
+        "email": None,
+        "summary": None,
+        "description": None,
+        "homepage": None,
+        "license": None,
+        "dependencies": None
+        }
+
+    for individual in file_data:
+        # update the value of name
+        if '.name ' in individual and 'basename' not in individual:
+            # >>> s = 'spec.name = "abc"'
+            # >>> name = s.split('= ')
+            # >>> name
+            # ['spec.name ', '"abc"']
+            # >>> name = name[1].strip('\'"%q{}')
+            # >>> name
+            # abc
+            name = individual.split('= ')
+            name = name[1].strip('\'"%q{}')
+            gemspec_data['name'] = name
+            continue
```

Comment:
What are the continues for here? I forgot if I asked this in the past, but to me they seem a bit ugly. Is there a better way to do this?

Reply:
I don't the better way but I am removing continue in this :)